File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int f(int n) {
    if(n <= 1) return 1;
    return n * f(n - 1);
}

int solution(int n) {
    int answer = 0;
    for(int i = 0; i <= n; i++) {
        if(f(i) <= n) {
            answer = i;
        } else break;
    }
    return answer;
}

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.swift
import Foundation

func f(_ n: Int) -> Int {
    if (n <= 1) { return 1 }
    return n * f(n - 1)
}

func solution(_ n:Int) -> Int {
    var num = 0, count = 1
    while num <= n {
        num = f(count)
        count += 1
    }
    return count - 2
}